ansible_sdk/_util/metrics.py

```diff
@@ -0,0 +1,207 @@
+                                "task_count": task_count,
+                                "task_duration": "",
+                            }
+                        )
+
+                    # Write job metrics data
+                    await job_csv_writer.writerow_async(
+                        {
+                            "job_id": ev["runner_ident"],
+                            "job_type": "local",
+                            "started": metrics_data.started,
+                            "finished": end_time,
+                            "job_state": "",
+                            "hosts_ok": len(ev["event_data"]["ok"]),
+                            "hosts_changed": len(ev["event_data"]["changed"]),
+                            "hosts_skipped": len(ev["event_data"]["skipped"]),
+                            "hosts_failed": len(ev["event_data"]["failures"]),
+                            "hosts_unreachable": len(ev["event_data"]["failures"]),
+                            "task_count": metrics_data.task_counter,
+                            "task_duration": end_time - metrics_data.started,
+                        }
+                    )
+
+                    # Write collection metrics data
+                    for collection, collection_count in metrics_data.collection_data.items():
+                        await collection_csv_writer.writerow_async(
+                            {
+                                "job_id": ev["runner_ident"],
+                                "collection_fqcn": collection,
+                                "collection_version": "",
+                                "task_count": collection_count,
+                                "task_duration": "",
+                            }
+                        )
+
+                    # Write role metrics data
+                    for role, role_count in metrics_data.role_data.items():
+                        await role_csv_writer.writerow_async(
+                            {
+                                "job_id": ev["runner_ident"],
+                                "role_fqcn": role,
+                                "collection_version": "",
+                                "task_count": role_count,
+                                "task_duration": "",
+                            }
+                        )
+            status_obj.drop_event(ev)
+
+        # Create a tar file for further consumption
+        metrics_tar_filename = os.path.join(
+            status_obj._job_def.metrics_output_path,
+            "%s_%s_job_data.tar.gz"
+            % (datetime.strftime(end_time, "%Y_%m_%d_%H_%M_%S"), runner_ident),
+        )
+
+        datafiles = [
+            job_csv_filename,
+            module_csv_filename,
+            collection_csv_filename,
+            roles_csv_filename,
+        ]
+
+        with tarfile.open(metrics_tar_filename, "w") as tfh:
+            for filename in datafiles:
+                tfh.add(filename)
```

Comment:
This needs to use the `arcname` arg with something like `os.path.basename(filename)` to avoid adding the containing dirs to the output tarball